deployment/promotion/promote.py

```diff
@@ -9,396 +9,157 @@
-            return
-        model = models[0]
-        print("Using existing registered model '%s'" % model['name'])
-    model_version = create_model_version(dest_auth, promotion['model_version'], model)
-
-    artifacts_and_model = promotion['artifacts']
-    artifacts_and_model.append(model_version['model'])
-    artifact_paths = upload_artifacts(dest_auth, model_version['id'], artifacts_and_model)
-
-    # Standard RMVs will have artifact path 'model' while ER->RMVs will have artifact path 'model.pkl'
-    model_artifact = model_version['model']
-    if 'model' in artifact_paths.keys():
-        model_artifact['path'] = artifact_paths['model']
-    else:
-        model_artifact['path'] = artifact_paths['model.pkl']
-    patch_model(dest_auth, model['id'], model_version['id'], model_artifact)
-
-    dest_build = create_build(dest_auth, model_version['id'], build_location, promotion['build'])
-    update_build_message(dest_auth, promotion['build'], dest_build)
-    return dest_build
-
-
-def promote(_config):
-    promotion = get_promotion_data(_config)
-    promoted_build = create_promotion(_config, promotion)
-    print("Promoted build %d created." % promoted_build['id'])
-
-
-promote(config)
+    print("Fetching or creating destination registered model %s" % source_model.name)
+    dest_model = dest_client.get_or_create_registered_model(name=source_model.name)
+
+# Copy information from the source model to the destination model
+# TODO: add readme, description to the client
+print("Copying info from source model to destination model")
+dest_model.add_labels([v for v in source_model.get_labels()])
+
+# Create the basic model version
+# TODO: add readme, description to the client
+# TODO: check return types
+print("Copying info from source model version to destination model version")
+dest_model_version = dest_model.create_version(
+    name=source_model_version.name,
+    labels=[v for v in source_model_version.get_labels()],
+    attrs={k: v for k, v in source_model_version.get_attributes().items()},
+)
+
+for key in source_model_version.get_artifact_keys():
+    artifact = source_model_version.get_artifact(key)
+    dest_model_version.log_artifact(key, artifact, overwrite=True)
+
+model = source_model_version.get_model()
+dest_model_version.log_model(model, custom_modules=[], overwrite=True)
+
+env = source_model_version.get_environment()
+dest_model_version.log_environment(env, overwrite=True)
+
+print("Copying build information")
+source_builds = source_model_version.list_builds()
+if len(source_builds) == 0:
+    raise ValueError("No builds found for model version %s" % source_model_version_id)
+source_build = source_builds[0]
+
+dest_build = dest_model_version.create_external_build(
+    location=source_build.location,
+    requires_root=source_build.requires_root,
+    scan_external=source_build.scan_external,
+    self_contained=source_build.self_contained,
+)
+dest_build.message = source_build.message
```

Comment:
I don't think this actually uploads the message to the BE; I think this is client-only

Reply:
ah, you added it. Nice